File: university-service/src/main/java/ua/com/foxminded/courseproject/repository/StudentRepository.java
package ua.com.foxminded.courseproject.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import ua.com.foxminded.courseproject.entity.Student;

import java.time.LocalDate;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface StudentRepository extends CrudRepository<Student, UUID> {

    Optional<Student> findById(UUID id);

    Page<Student> findAll(Pageable pageable);

    Student save(Student student);

    void delete( Student student);

    Boolean existsStudentByFirstNameAndLastNameAndBirthDay(String firstName, String lastName, LocalDate birthDay);
}

File: university-service/src/main/kotlin/ua/com/foxminded/courseproject/repository/StudentRepository.kt
package ua.com.foxminded.courseproject.repository

import java.time.LocalDate
import java.util.UUID
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ua.com.foxminded.courseproject.entity.Student

interface StudentRepository {
    fun findById(id: UUID): Mono<Student>
    fun findAll(): Flux<Student>

    fun save(student: Student): Mono<Student>
    fun delete(student: Student): Mono<Void>
    fun existsStudentByFirstNameAndLastNameAndBirthDay(
        firstName: String,
        lastName: String,
        birthDay: LocalDate
    ): Mono<Boolean>
}

